File: BIT_MANIPULATION/duplicate_el.cpp
#include <iostream>
using namespace std;

// Given an arr of length n, return the element which is repeated twice
// given the fact that all the elements occour once, except 1;
int number(int* arr, int n)
{
    int ans = 0;
    for ( int i=0; i<n; i++ )
    {
        ans = ans ^ arr[i];
    }
    return ans;
}

int main()
{
    int arr[] = {1, 2, 3, 4, 5, 6, 7, 7};
    int n = sizeof(arr)/sizeof(arr[0]);
    cout << number(arr, n);
    return 0;
}

File: BIT_MANIPULATION/duplicate_el.py
from functools import reduce
# given an array of n integers, where all elements occur twice, except one element which occurs only once. Find that element.


def ans(arr: list) -> int:
    return reduce(lambda x, y: x ^ y, arr)


print(ans([1, 1, 2, 3, 2, 3, 5, 5, 9]))
